src/main/js/providers/ProviderTable.js

```diff
@@ -0,0 +1,363 @@
+            this.state.saveCallback(true);
+        }
+    }
+
+    combineModelWithDefaults(providerConfig) {
+        const { descriptor } = this.state;
+        if (!descriptor) {
+            return {};
+        }
+        const emptyConfig = FieldModelUtilities.createFieldModelWithDefaults(descriptor.fields, DescriptorUtilities.CONTEXT_TYPE.GLOBAL, descriptor.name);
+        const updatedFieldModel = FieldModelUtilities.combineFieldModels(emptyConfig, providerConfig);
+        if (providerConfig.id) {
+            updatedFieldModel.id = providerConfig.id;
+        }
+        return updatedFieldModel;
+    }
+
+    createColumns() {
+        return [
+            {
+                header: 'id',
+                headerLabel: 'Id',
+                isKey: true,
+                hidden: true
+            },
+            {
+                header: 'name',
+                headerLabel: 'Name',
+                isKey: false,
+                hidden: false
+            },
+            {
+                header: 'createdAt',
+                headerLabel: 'Created At',
+                isKey: false,
+                hidden: false
+            },
+            {
+                header: 'lastUpdated',
+                headerLabel: 'Last Updated',
+                isKey: false,
+                hidden: false
+            },
+            {
+                header: 'enabled',
+                headerLabel: 'Enabled',
+                isKey: false,
+                hidden: false,
+                dataFormat: this.enabledState
+            }
+        ];
+    }
+
+    enabledState(cell) {
+        const icon = (cell == 'true') ? 'check' : 'times';
+        const color = (cell == 'true') ? 'synopsysGreen' : 'synopsysRed';
+        const className = `alert-icon ${color}`;
+
+        return (
+            <div className="btn btn-link jobIconButton">
+                <FontAwesomeIcon icon={icon} className={className} size="lg" />
+            </div>
+        );
+    }
+
+    onConfigClose(callback) {
+        this.props.clearFieldErrors();
+        callback();
+    }
+
+    clearModalFieldState() {
+        if (this.state.providerConfig && Object.keys(this.state.providerConfig).length > 0) {
+            this.setState({
+                providerConfig: {}
+            });
+        }
+    }
+
+    retrieveData() {
+        const { descriptor } = this.state;
+        if (descriptor) {
+            this.props.getAllConfigs(descriptor.name);
+        }
+    }
+
+    handleChange(e) {
+        const { name, value, type, checked } = e.target;
+        const { providerConfig } = this.state;
+
+        const updatedValue = type === 'checkbox' ? checked.toString().toLowerCase() === 'true' : value;
+        const newProviderConfig = Object.assign(providerConfig, FieldModelUtilities.updateFieldModelSingleValue(providerConfig, name, updatedValue));
+
+        this.setState({
+            providerConfig: newProviderConfig
+        });
+    }
```

Comment:
HandleChange seems to be very similar everywhere. Could this be made common somewhere? If it's similar to the TableDisplay, perhaps we add most of this code there?

Reply:
FieldsPanel handles its on handleChange similar to this. Perhaps we could do it that way.

Reply:
I will look into this with another feature I will add.  I will consider it for a different PR.